File: jambeth/jAmbeth.Service/src/main/java/de/osthus/ambeth/merge/IValueObjectConfig.java
package de.osthus.ambeth.merge;

public interface IValueObjectConfig
{
	Class<?> getEntityType();

	Class<?> getValueType();

	String getValueObjectMemberName(String businessObjectMemberName);

	ValueObjectMemberType getValueObjectMemberType(String valueObjectMemberName);

	boolean holdsListType(String memberName);

	boolean isIgnoredMember(String memberName);

	Class<?> getMemberType(String memberName);
}

File: ambeth/Ambeth.Service/ambeth/merge/IValueObjectConfig.cs
using System;

namespace De.Osthus.Ambeth.Merge
{
    public interface IValueObjectConfig
    {
        Type EntityType { get; }

        Type ValueType { get; }

        String GetValueObjectMemberName(String businessObjectMemberName);

        ValueObjectMemberType GetValueObjectMemberType(String valueObjectMemberName);

        bool HoldsListType(String memberName);

        bool IsIgnoredMember(String memberName);

        Type GetMemberType(String memberName);
    }
}
